src/System.Text.Json/src/System/Text/Json/JsonDocument.TryGetProperty.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+
+namespace System.Text.Json
+{
+    public sealed partial class JsonDocument
+    {
+        internal bool TryGetNamedPropertyValue(int index, ReadOnlySpan<char> propertyName, out JsonElement value)
+        {
+            if (_utf8Json.IsEmpty)
+                throw new ObjectDisposedException(nameof(JsonDocument));
+
+            _parsedData.Get(index, out DbRow row);
+
+            if (row.TokenType != JsonTokenType.StartObject)
+            {
+                throw new InvalidOperationException();
+            }
+
+            if (row.NumberOfRows == 0)
+            {
+                value = default;
+                return false;
+            }
+
+            int maxBytes = Encoding.UTF8.GetMaxByteCount(propertyName.Length);
+            int endIndex = checked(row.NumberOfRows * DbRow.Size + index);
+
+            // The biggest number of bytes we're willing to pre-UTF8
+            const int StackUtf8Max = 256;
+
+            if (maxBytes < StackUtf8Max)
+            {
+                Span<byte> utf8Name = stackalloc byte[StackUtf8Max];
+                int len = Encoding.UTF8.GetBytes(propertyName, utf8Name);
+                utf8Name = utf8Name.Slice(0, len);
+
+                return TryGetNamedPropertyValue(
+                    index + DbRow.Size,
+                    endIndex,
+                    utf8Name,
+                    out value);
+            }
+
+            int minBytes = propertyName.Length;
+            int candidateIndex = index + DbRow.Size;
+
+            while (candidateIndex < endIndex)
+            {
+                _parsedData.Get(index, out row);
+                Debug.Assert(row.TokenType == JsonTokenType.String);
+
+                if (row.SizeOrLength >= minBytes && row.SizeOrLength <= maxBytes)
+                {
+                    byte[] tmpUtf8 = ArrayPool<byte>.Shared.Rent(maxBytes);
+                    Span<byte> utf8Name = default;
+
+                    try
+                    {
+                        int len = Encoding.UTF8.GetBytes(propertyName, tmpUtf8);
+                        utf8Name = tmpUtf8.AsSpan(0, len);
+
+                        return TryGetNamedPropertyValue(
+                            candidateIndex,
+                            endIndex,
+                            utf8Name,
+                            out value);
+                    }
+                    finally
+                    {
+                        utf8Name.Clear();
```

Comment:
How come?

Reply:
'cuz I'm a security dev and hate when people can see my uncleared memory :smile:  I suppose it isn't really needed here, since the property names themselves aren't usually a secret.